jax/_src/interpreters/batching.py

```diff
@@ -143,6 +147,22 @@ def make_batch_axis(
   else:
     return canonicalize_axis(stacked_axis, ndim)
 
+def bdim_as_shape(bdim, data_shape):
```

Comment:
Type annotations? Does Python even work without type annotations on function definitions!?

Reply:
Per contra in Haskell, the type annotations are optional but the program is always type-correct anyway.

Added.